crates/rome_bin/src/service/mod.rs

```
@@ -0,0 +1,195 @@
+                                io::ErrorKind::UnexpectedEof,
+                                "unexpected EOF",
+                            )));
+                        }
+                        // A read of two bytes corresponds to the "\r\n" sequence
+                        // that indicates the end of the header section
+                        2 => break,
+                        _ => {
+                            let colon = line.find(':').ok_or_else(|| {
+                                RomeError::IoError(io::Error::new(
+                                    io::ErrorKind::Other,
+                                    "invalid header line",
+                                ))
+                            })?;
+
+                            let (name, value) = line.split_at(colon);
+                            let value = value[1..].trim();
+
+                            match name {
+                                "Content-Length" => {
+                                    length = Some(value.parse().map_err(|err| {
+                                        RomeError::IoError(io::Error::new(
+                                            io::ErrorKind::Other,
+                                            err,
+                                        ))
+                                    })?);
+                                }
+                                "Content-Type" => {
+                                    if !value.starts_with("application/json") {
+                                        return Err(RomeError::IoError(io::Error::new(
+                                            io::ErrorKind::Other,
+                                            "unsupported Content-Type encoding",
+                                        )));
+                                    }
+                                }
+                                _ => {
+                                    // Ignore unknown headers
```

It may be helpful to log unknown headers or someone might have a very frustrating experience when they e.g. have a typo in 'Content-Type'

---

I added an `eprintln` call there, but in practice the `SocketTransport` should only be used to connect to a `rome_lsp` instance
and this branch should only ever be run if there's a version mismatch between the running daemon and the CLI (this is also the case for the `Content-Type` header, the current version of the server doesn't send it since its assumed to be JSON by default, but while I envision unsupported headers can simply be ignored by the CLI the socket should return an error if the server sends in a response with an encoding it doesn't support)